es-core/src/PowerSaver.cpp

```diff
@@ -63,6 +73,11 @@ void PowerSaver::setState(bool state)
 void PowerSaver::runningScreenSaver(bool state)
 {
 	mRunningScreenSaver = state;
+	if (mWakeupTimeout < mMode)
+	{
+		// Disable PS if wake up time is less than mode as PS will never trigger
+		setState(!state);
```

Comment:
Granted, I don't know enough about the PS logic, but are we not being unnecessarily smart here, and potentially creating an error condition here?

I believe that, if you are running the screensaver, this does what we want (sets state to false if it's lower than the mode value), but if we are not running the screensaver and won't this make it behave as if it is running the screensaver when it shouldn't?

Reply:
`PowerSaver::runningScreenSaver(bool state)` is called only when triggering screensaver with wake up support, video and image, and not for dim and blank.